### PR TITLE
fix: Replace structure as service

### PR DESCRIPTION
**Pull request by BenjaminMichaelis**
Fixes IntelliTect-dev/EssentialCSharp.Tooling#742

### CODE REVIEW

File: EssentialCSharp.Web/Services/ISiteMappingService.cs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+﻿namespace EssentialCSharp.Web.Services
```
Comment (Keboo) [inline EssentialCSharp.Web/Services/ISiteMappingService.cs:1]: File scoped namespace

File: EssentialCSharp.Web/Services/SiteMappingService.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+﻿using Microsoft.Extensions.Options;
+
+namespace EssentialCSharp.Web.Services
+{
+    public class SiteMappingService
```
Comment (Keboo) [inline EssentialCSharp.Web/Services/SiteMappingService.cs:5]: Should this implement the interface?

File: EssentialCSharp.Web/Services/SiteMappingService.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    {
+        public IList<SiteMapping> SiteMappings { get; init; }
```
Comment (Keboo) [inline EssentialCSharp.Web/Services/SiteMappingService.cs:7]:
```suggestion
        public IList<SiteMapping> SiteMappings { get; }
```

File: EssentialCSharp.Web/Controllers/HomeController.cs [hunk 4]
```diff
@@ -7,14 +8,16 @@ namespace EssentialCSharp.Web.Controllers;
 public class HomeController : Controller
 {
     private readonly IWebHostEnvironment _HostingEnvironment;
-    private readonly IList<SiteMapping> _SiteMappings;
+    private readonly SiteMappingService _SiteMappingService;
     private readonly ILogger<HomeController> _Logger;
+    private readonly IList<SiteMapping> _SiteMappings;
```
Comment (Keboo) [inline EssentialCSharp.Web/Controllers/HomeController.cs:13]: Do you need this list? I would just store the service and go back to it every time you need the list.

File: EssentialCSharp.Web/Services/SiteMappingService.cs (hunk 2 above)
Comment (BenjaminMichaelis) [reply EssentialCSharp.Web/Services/SiteMappingService.cs:5]: Which interface?

Comment (Keboo) [reply EssentialCSharp.Web/Services/SiteMappingService.cs:5]: You created a `ISiteMappingService`

Comment (BenjaminMichaelis) [reply EssentialCSharp.Web/Services/SiteMappingService.cs:5]: What advantage does that provide here?

File: EssentialCSharp.Web/Services/SiteMappingService.cs [hunk 5]
```diff
@@ -0,0 +1,17 @@
+﻿namespace EssentialCSharp.Web.Services;
+
+public class SiteMappingService
```
Comment (Keboo) [inline EssentialCSharp.Web/Services/SiteMappingService.cs:3]: We should probably put this behind an interface for testability. Related we should add some integration test for that controller.

Comment (BenjaminMichaelis) [reply EssentialCSharp.Web/Services/SiteMappingService.cs]: Opened https://github.com/IntelliTect-dev/EssentialCSharp.Tooling/issues/881 for the testing part

File: EssentialCSharp.Web/Services/ISiteMappingService.cs [hunk 6]
```diff
@@ -0,0 +1,5 @@
+﻿namespace EssentialCSharp.Web.Services;
+
+public interface ISiteMappingService
+{
```
Comment (Keboo) [inline EssentialCSharp.Web/Services/ISiteMappingService.cs:4]:
```suggestion
{
    IList<SiteMapping> SiteMappings { get; }
```

File: EssentialCSharp.Web/Program.cs [hunk 7]
```diff
@@ -9,20 +11,9 @@ private static void Main(string[] args)
...
-            IWebHostEnvironment _HostingEnvironment = sp.GetRequiredService<IWebHostEnvironment>();
-            string path = Path.Combine(_HostingEnvironment.ContentRootPath, "Chapters", "sitemap.json");
-            List<SiteMapping>? siteMappings = System.Text.Json.JsonSerializer.Deserialize<List<SiteMapping>>(File.OpenRead(path));
-            if (siteMappings is null)
-            {
-                throw new InvalidOperationException("No table of contents found");
-            }
-            return siteMappings;
-        });
+        builder.Services.AddSingleton<SiteMappingService>();
```
Comment (Keboo) [inline EssentialCSharp.Web/Program.cs:14]:
```suggestion
        builder.Services.AddSingleton<ISiteMappingService, SiteMappingService>();
```

File: EssentialCSharp.Web/Controllers/HomeController.cs [hunk 8]
```diff
@@ -7,20 +8,20 @@ namespace EssentialCSharp.Web.Controllers;
 public class HomeController : Controller
 {
     private readonly IWebHostEnvironment _HostingEnvironment;
-    private readonly IList<SiteMapping> _SiteMappings;
+    private readonly SiteMappingService _SiteMappingService;
     private readonly ILogger<HomeController> _Logger;
 
-    public HomeController(ILogger<HomeController> logger, IWebHostEnvironment hostingEnvironment, IList<SiteMapping> siteMappings)
+    public HomeController(ILogger<HomeController> logger, IWebHostEnvironment hostingEnvironment, SiteMappingService siteMappingService)
```
Comment (Keboo) [inline EssentialCSharp.Web/Controllers/HomeController.cs:14]:
```suggestion
    public HomeController(ILogger<HomeController> logger, IWebHostEnvironment hostingEnvironment, ISiteMappingService siteMappingService)
```

File: EssentialCSharp.Web/Controllers/HomeController.cs [hunk 9]
```diff
@@ -7,20 +8,20 @@ namespace EssentialCSharp.Web.Controllers;
 public class HomeController : Controller
 {
     private readonly IWebHostEnvironment _HostingEnvironment;
-    private readonly IList<SiteMapping> _SiteMappings;
+    private readonly SiteMappingService _SiteMappingService;
```
Comment (Keboo) [inline EssentialCSharp.Web/Controllers/HomeController.cs:11]:
```suggestion
    private readonly ISiteMappingService _SiteMappingService;
```

**Review by Keboo — APPROVED** [review]

**Pull request merged**